movieHW/movieHW/Views/MoviesViewController.swift

```diff
@@ -0,0 +1,248 @@
+            .isActive = true
+        comingSoonButton.widthAnchor.constraint(equalToConstant: 100.0).isActive = true
+        comingSoonButton.heightAnchor.constraint(equalToConstant: 50.0).isActive = true
+        comingSoonButton.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 10).isActive = true
+        comingSoonButton.setTitleColor(UIColor.white, for: .normal)
+
+        comingSoonButton.layer.cornerRadius = 9
+        comingSoonButton.layer.borderColor = UIColor.white.cgColor
+        comingSoonButton.layer.borderWidth = 1.0
+
+        comingSoonButton.addTarget(self, action: #selector(comingSoonButtonAction), for: .touchUpInside)
+    }
+
+    private func createTableViewSettings() {
+        tableView.addSubview(refresherControl)
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.frame = view.frame
+        tableView.register(
+            MovieTableViewCell.self,
+            forCellReuseIdentifier: CellIdentifier.movieCellIdentifier
+        )
+
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor, constant: 0).isActive = true
+        tableView.rightAnchor.constraint(equalTo: view.safeAreaLayoutGuide.rightAnchor, constant: 0).isActive = true
+        tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 75).isActive = true
+        tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: 0).isActive = true
+        tableView.separatorStyle = .none
+    }
+
+    private func loadMovies(urlString: String) {
+        request.getMovies(
+            urlString: urlString
+        ) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case let .failure(error):
+                print(error.localizedDescription)
+            case let .success(movies):
+                self.movies = movies
+            }
+        }
+    }
+
+    private func configureView() {
+        view.addSubview(tableView)
+        view.backgroundColor = UIColor(named: Colors.defaultBlack)
+    }
+
+    private func setupNavigationBar() {
+        let backButtonItem = UIBarButtonItem()
+        backButtonItem.tintColor = .orange
+        backButtonItem.title = Constants.backBarButtonTitleText
+
+        navigationItem.backBarButtonItem = backButtonItem
+    }
+}
+
+// MARK: - UITableViewDelegate, UITableViewDataSource
+
+extension MoviesViewController: UITableViewDelegate, UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        movies.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        1
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let detailVC = DetailViewController()
+        detailVC.createImages(
```

Comment:
```suggestion
        detailVC.createImages(movie: movies[indexPath.section])
```
 передаем структуру, зачем нам каждый параметр дергать из нее отдельно в методе?